apps/spruce/src/components/TaskStatusIconLegend/index.tsx

```diff
@@ -1,109 +1,158 @@
+  gap: ${size.xs};
+`;
+
+const WaterfallContainer = styled.div`
+  width: 420px;
+  height: 150px;
+  display: flex;
+  flex-direction: column;
+  flex-wrap: wrap;
+  gap: ${size.xs};
+`;
+
+const Row = styled.div`
+  display: flex;
+  align-items: flex-start;
+  gap: ${size.xxs};
+`;
+
+const TaskStatusIcon = styled.div`
+  flex-shrink: 0;
+`;
 
-export const TaskStatusIconLegend: React.FC = () => {
-  const { sendEvent } = useProjectHealthAnalytics({ page: "Commit chart" });
-  const [isActive, setIsActive] = useState(false);
+const LabelContainer = styled.div``;
+
+type TaskStatusIconLegendProps = {
+  useWaterfall: boolean;
+};
+
+export const TaskStatusIconLegend: React.FC<TaskStatusIconLegendProps> = ({
+  useWaterfall,
+}) => {
+  const { sendEvent } = (
+    useWaterfall ? useWaterfallAnalytics : useProjectHealthAnalytics
+  )({ page: "Commit chart" });
+
+  const isMainlineCommits = !!useMatch(`${routes.commits}/*`);
+  const isWaterfall = !!useMatch(`${routes.waterfall}/*`);
+
+  const [open, setOpen] = useState(false);
+
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  const popoverRef = useRef<HTMLDivElement>(null);
+
+  useOnClickOutside([buttonRef, popoverRef], () => setOpen(false));
 
   return (
     <div>
       <IconButton
-        aria-label="Task Status Icon Legend"
+        ref={buttonRef}
+        aria-label="Task status icon legend"
         onClick={() => {
-          setIsActive(!isActive);
           sendEvent({
             name: "Toggled task icon legend",
-            open: true,
+            open: !open,
           });
+          setOpen(!open);
         }}
       >
-        <StyledIcon glyph="QuestionMarkWithCircle" />
+        <Icon glyph="QuestionMarkWithCircle" />
       </IconButton>
       <Popover
-        active={isActive}
-        align="top"
-        justify="end"
-        // In some cases, the z-index of the popover needs to be higher than the rest
-        // of the app due to some components having a higher z-index.
-        popoverZIndex={zIndex.tooltip}
-        usePortal
+        ref={popoverRef}
+        active={open}
+        align={Align.Top}
+        justify={Justify.End}
+        popoverZIndex={zIndex.popover}
+        refEl={buttonRef}
       >
         <StyledPopoverContainer>
           <TitleContainer>
             <Overline>Icon Legend</Overline>
             <IconButton
-              aria-label="Close Task Status Icon Legend"
+              aria-label="Close task status icon legend"
               onClick={() => {
                 sendEvent({
                   name: "Toggled task icon legend",
                   open: false,
                 });
-                setIsActive(false);
+                setOpen(false);
               }}
             >
               <Icon glyph="X" />
             </IconButton>
           </TitleContainer>
-          <LegendContent />
+          <LegendContent
+            useWaterfall={isWaterfall || (useWaterfall && !isMainlineCommits)}
```

Comment:
If I'm not mistaken the build variant and task history pages use the old icon set. Maybe for now we should only show the new icon set on the waterfall page and default to the old set. When the task history page is redone we can update the legend there too.

Reply:
ok sounds good!